benchmarks/r_csv_read_benchmark.py

```diff
@@ -0,0 +1,60 @@
+import itertools
```

Comment:
Since we do have a Python csv benchmark already, we should probably put that in the same file under `benchmarks/csv_read_benchmark.py`

We also don't _need_ to use this if it's more work than helpful, but there is a class `_benchmark.BenchmarkPythonR` which should do both at the same time without needing to repeat a number of the same boiler plate code. Though IIRC, arrowbench doesn't have a streaming reading option, so we would need to disable that for the R side — and possibly a few other slight differences or changes.

https://github.com/ursacomputing/benchmarks/blob/b252842cc629fdea4eb8eec2b9db9f1eb7324040/benchmarks/file_benchmark.py#L27-L54

Reply:
Yeah, I found that because the R tests for it are failing locally for me (but not in CI) and TPC-H R ones are failing on CI (but not locally) because it can't find {remotes} even though it installs it. (Sam is running into similar issue right now I believe.)

I'm just going to make a new branch and see if I can switch the existing one to `BenchmarkPythonR`, but I suspect the tests are still going to fail, maybe worse—something bigger is going on. But maybe they'll fail differently?

Reply:
Sounds good — let me know if there's an approve you need on that for CI to run.

And yeah, the {remotes} issue is frustrating! Is there an issue already for that? maybe we can solve it separately?

Reply:
I don't think there's a story yet, but I'm not sure how to reproduce it; I've only seen it in gh actions

Reply:
Oh, I think I know what's going on, I'll make a PR one sec

Reply:
Ok, I think this is resolved now — I ended up making a slight change to arrowbench, but I'm seeing it pass at: https://github.com/ursacomputing/benchmarks/runs/6118529147?check_suite_focus=true

I suspect something might have slightly changed in github actions + R where the libpaths where we installed weren't carrying through exactly, but being explicit seems to have worked!

Reply:
Nice, thank you! I'm getting close on a refactored version using `BenchmarkPythonR`; I'll pull arrowbench and test with that before pushing.